### ISSUE TITLE
Selection defect related to select-half-plane

### ISSUE THREAD

**Issue by vorth** [vZome/vzome#263]
I selected a panel, then attempted "select half plane".  The error dialog explained what was wrong, and I dismissed it.  Then I clicked on a strut by accident (missing the background, to deselect all), and this error occurred... or perhaps when I tried again to deselect all.

```
WARNING: This command requires four selected connectors.

The first three connectors must not be collinear,
so that they define a plane.
The fourth connector must lie outside of that plane,
and defines which half space you wish to select.

All parts that are completely within that half-space will be selected.

You have selected fewer than four connectors.
INFO: command failure
WARNING: java.util.NoSuchElementException
java.util.NoSuchElementException
        at com.vzome.core.generic.FilteredIterator.next(FilteredIterator.java:95)
        at org.vorthmann.zome.app.impl.MeasureController.selectionChanged(MeasureController.java:81)
        at com.vzome.core.editor.SelectionSummary.notifyListeners(SelectionSummary.java:30)
        at com.vzome.core.editor.EditorModelImpl.notifyListeners(EditorModelImpl.java:196)
        at com.vzome.core.editor.DocumentModel$1.publishChanges(DocumentModel.java:285)
        at com.vzome.core.editor.EditHistory.undo(EditHistory.java:257)
        at org.vorthmann.zome.app.impl.UndoRedoController.doAction(UndoRedoController.java:43)
        at org.vorthmann.ui.DefaultController.actionPerformed(DefaultController.java:44)
        at org.vorthmann.zome.ui.ControllerActionListener.actionPerformed(ControllerActionListener.java:21)
        at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
        at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
        at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
        at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
        at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
        at javax.swing.AbstractButton.doClick(AbstractButton.java:356)
        at javax.swing.plaf.basic.BasicMenuItemUI$Actions.actionPerformed(BasicMenuItemUI.java:811)
        at javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1668)
        at javax.swing.JComponent.processKeyBinding(JComponent.java:2882)
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(JMenuBar.java:699)
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(JMenuBar.java:706)
        at javax.swing.JMenuBar.processBindingForKeyStrokeRecursive(JMenuBar.java:706)
        at javax.swing.JMenuBar.processKeyBinding(JMenuBar.java:677)
        at javax.swing.KeyboardManager.fireBinding(KeyboardManager.java:307)
        at javax.swing.KeyboardManager.fireKeyboardAction(KeyboardManager.java:293)
        at javax.swing.JComponent.processKeyBindingsForAllComponents(JComponent.java:2974)
        at javax.swing.JComponent.processKeyBindings(JComponent.java:2966)
        at javax.swing.SwingUtilities.processKeyBindings(SwingUtilities.java:1585)
        at javax.swing.UIManager$2.postProcessKeyEvent(UIManager.java:1483)
        at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:845)
        at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1102)
        at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:973)
        at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:799)
        at java.awt.Component.dispatchEventImpl(Component.java:4760)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
        at java.awt.EventQueue$4.run(EventQueue.java:733)
        at java.awt.EventQueue$4.run(EventQueue.java:731)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
WARNING: internal error: null
INFO: command failure
WARNING: java.util.NoSuchElementException
INFO: command failure
WARNING: java.lang.NullPointerException
```

**Comment by david-hall** [comment]
I ran into pretty much this exact same scenario tonight. It's easy to reproduce using a model with just enough balls and struts to make a panel. The NoSuchElementException keeps happening every time I click on anything after having invoked SelectByPlane with just the panel selected. I came here to report the bug and found that it's already been reported.

**Comment by david-hall** [comment]
As a side note, I think that SelectByPlane should also accept a panel and a non-collinear ball as arguments sine that's more intuitive, but that's a feature request, not a bug. I think we need to get this bug fixed before making that change.

**Comment by david-hall** [comment]
Closing this issue. Fixed by commit 7ee21295865de5dfb981292863767e401562bea5.
